crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -949,44 +949,80 @@ impl<'src> Parser<'src> {
         }
     }
 
+    /// Parse a comparison expression.
+    ///
+    /// This includes the following operators:
+    /// - Value comparisons: `==`, `!=`, `<`, `<=`, `>`, and `>=`.
+    /// - Membership tests: `in` and `not in`.
+    /// - Identity tests: `is` and `is not`.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at any of the comparison operators.
+    ///
+    /// See: <https://docs.python.org/3/reference/expressions.html#comparisons>
     fn parse_compare_expression(
         &mut self,
         lhs: Expr,
         start: TextSize,
-        op: TokenKind,
-        op_bp: Precedence,
+        operator: TokenKind,
+        operator_binding_power: Precedence,
     ) -> ast::ExprCompare {
-        let mut comparators = vec![];
-        let op = token_kind_to_cmp_op([op, self.current_token_kind()]).unwrap();
-        let mut ops = vec![op];
+        let compare_operator = token_kind_to_cmp_op([operator, self.current_token_kind()]).unwrap();
 
```

Comment:
Nit: I think I would change the signature and call site to avoid having to pass both `operator`, `operator_binding_power` AND convert `operator` to `CmpOp`

1. Add a new `as_compare_operator` method to `TokenKind` that returns `Option<CompareOp>`
1. Refactor `TokenKind::is_compare_operator` to `self.as_compare_operator().is_some()`
1. `parse_expression_with_precedence`: Replace `op.is_compare_operator` with `if let Some(compare_operator) = op.as_compare_operator() {` 
1. Implement `From<CmpOp> for Precedence` (can never fail)
1. Change `parse_compare_expression` to only accept a `CmpOp`
  * You can get the token by calling the (inlineable) `TokenKind::from(cmp_op)` 
  * You can get the precedence by calling the (inlineable) `Precedence::from(cmp_op)


Edit: Okay, `as_compare_operator` may not work because it requires lookahead. I think we can either keep `as_compare_operator` and match on the comparator in the parser to see if we may need to change it in case it's a two kind operator (we need to bump the extra token anyway)